build/tools/SharedCodeGenerator/Outputs/CSharpOutput.cs

```diff
@@ -58,7 +58,11 @@ public string GetContent()
                 Name = this.className,
                 Scope = scope,
                 StringConstants = this.collection.StringConstants?.ToDictionary(pair => pair.Key.Camelize(), pair => pair.Value),
-                ListConstants = this.collection.ListConstants?.ToDictionary(pair => pair.Key.Camelize(), pair => $"{{ \"{string.Join("\", \"", pair.Value)}\" }}"),
+                ListConstants = this.collection.ListConstants?.ToDictionary(
+                    pair => pair.Key.Camelize(),
+                    pair => pair.Value.Any(s => !string.IsNullOrEmpty(s?.ToString()))
+                        ? $"{{ \"{string.Join("\", \"", pair.Value)}\" }}"
+                        : "{ }"),
```

Comment:
nit: just confirming that this handles the case where no item is provided to a list in `constants.yaml`

Reply:
yep, this addition makes it so that the list will be actually empty in the c sharp output if all of the list contents are empty

Reply:
Sweet -- thanks for the confirmation 👍